src/components/organisms/teacher/OakUnitListOptionalityItem/OakUnitListOptionalityItem.tsx

```diff
@@ -0,0 +1,244 @@
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      $mr={[null, "space-between-xxxl"]}
+    >
+      {yearTitle}
+    </OakSpan>
+
+    <OakBox
+      $ph={"inner-padding-xs"}
+      $pv={"inner-padding-ssx"}
+      $ba={"border-solid-s"}
+      $borderColor={
+        disabledOrUnavailable ? "border-neutral-lighter" : "bg-decorative4-main"
+      }
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      $background={
+        disabledOrUnavailable
+          ? "bg-neutral-stronger"
+          : "bg-decorative4-very-subdued"
+      }
+      $borderRadius={"border-radius-s"}
+      $ml={[null, "space-between-s"]}
+    >
+      <OakLabel
+        $font={"heading-light-7"}
+        $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      >{`${optionalityUnitsLength} unit options`}</OakLabel>
+    </OakBox>
+  </>
+);
+
+const StyledIndex = ({
+  index,
+  disabledOrUnavailable,
+  ...rest
+}: {
+  index: number;
+  disabledOrUnavailable: boolean | undefined;
+} & FlexStyleProps) => (
+  <OakFlex
+    $btlr={"border-radius-s"}
+    $bblr={"border-radius-s"}
+    $background={disabledOrUnavailable ? "bg-neutral-stronger" : "lavender"}
+    $minWidth={"all-spacing-11"}
+    $width={["all-spacing-11", "all-spacing-11", "auto"]}
+    $height={["all-spacing-11", "auto", "auto"]}
+    $justifyContent={"center"}
+    $alignItems={"center"}
+    {...rest}
+  >
+    <OakSpan
+      $font={"heading-5"}
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+    >
+      {index}
+    </OakSpan>
+  </OakFlex>
+);
+
+const StyledUnitHeading = ({
+  nullTitle,
+  disabledOrUnavailable,
+}: {
+  nullTitle: string | undefined;
+  disabledOrUnavailable: boolean | undefined;
+}) => {
+  if (nullTitle === undefined) {
+    return null;
+  }
+  return (
+    <OakFlex $maxWidth={"all-spacing-21"}>
+      <OakHeading
+        $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+        $font={"heading-7"}
+        tag="h3"
+      >
+        {nullTitle}
+      </OakHeading>
+    </OakFlex>
+  );
+};
+
+export type OakUnitListOptionalityItemProps<C extends ElementType> = {
+  as?: C;
+  disabled?: boolean;
+  unavailable?: boolean;
+  index: number;
+  nullTitle: string | undefined;
+  yearTitle?: string | null;
+
+  optionalityUnits: {
+    title: string;
+    href: string;
+
+    lessonCount: number;
+    firstItemRef?:
+      | MutableRefObject<HTMLAnchorElement | null>
+      | null
+      | undefined;
+  }[];
+} & ComponentPropsWithoutRef<C>;
+
+/**
+ *
+ * OakUnitsListOptionalityItem component used as links for unit cards with optionality
+ */
+export const OakUnitListOptionalityItem = <C extends ElementType>(
+  props: OakUnitListOptionalityItemProps<C>,
+) => {
+  const {
+    as,
+    lessonSectionName,
+    lessonCount,
+    progress,
+    disabled,
+    href,
+    unavailable,
+    onClick,
+    index,
+    firstItemRef,
+    optionalityUnits,
+    nullTitle,
```

Comment:
A bunch of these props aren't mentioned in the type for this component's props,

Are you doing this so that you can omit them after spreading them from your consumer in OWA or is this an oversight?